src/main/kotlin/com/ampnet/headlesscmsservice/service/impl/MailServiceImpl.kt

```diff
@@ -0,0 +1,117 @@
+package com.ampnet.headlesscmsservice.service.impl
+
+import com.ampnet.headlesscmsservice.enums.Lang
+import com.ampnet.headlesscmsservice.enums.MailFieldName
+import com.ampnet.headlesscmsservice.enums.MailType
+import com.ampnet.headlesscmsservice.exception.ErrorCode
+import com.ampnet.headlesscmsservice.exception.InvalidRequestException
+import com.ampnet.headlesscmsservice.exception.ResourceNotFoundException
+import com.ampnet.headlesscmsservice.persistence.respository.MailRepository
+import com.ampnet.headlesscmsservice.service.MailService
+import com.ampnet.headlesscmsservice.service.TranslationService
+import com.ampnet.headlesscmsservice.service.pojo.MailListResponse
+import com.ampnet.headlesscmsservice.service.pojo.MailResponse
+import com.ampnet.headlesscmsservice.service.pojo.MailTypeAndLang
+import com.ampnet.headlesscmsservice.service.pojo.MailUpdateServiceRequest
+import org.springframework.stereotype.Service
+import java.util.UUID
+
+@Service
+class MailServiceImpl(
+    private val mailRepository: MailRepository,
+    private val translationService: TranslationService
+) : MailService {
+
```

Comment:
It would be nice to comment this function because it is a bit hard to understand. 
```suggestion
    /**
     * Returns a list of mail responses for given coop by mail type and language.
     * If the mail is not defined, returns default value...
     *
     * @param coop String identifier of the cooperative
     * @param type optional [MailType]
     * @param lang optional [Lang]
     * @return a list of [MailResponse]
     */
```